Automattic/jetpack
# Introduce compose-extras.yml for custom mounting of volumes

Introduces ability to easily declare custom mounts of directories into the container.


#### Changes proposed in this Pull Request:


* Adds `docker/compose-extras.yml.sample` with sample lines for mounting a local dir into the container when it's started by `yarn docker:up`.
* Adds the package.json script `yarn docker:compose-extras` which will copy the sample file into a gitignored file `docker/compose-extras.yml`.
* Makes the command `yarn docker:compose` always load `docker/compose-extras.yml` for taking in account the customizations.
* Adds `docker/compose-extras.yml` to .gitignore so one can modify that file freely while coding without having its changes tracked

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

New feature but not user-facing.

#### Testing instructions:

1. Checkout this branch
2. Run `yarn docker:up` (hopefully having run `yarn docker:stop` beforehand.
3. Confirm that you have a new file `docker/compose-extras.yml`.
4. Run `yarn docker:stop`.
5. Update `docker/compose-extras.yml` to include the mounting of one plugin you actually have locally somewhere else. Save and close the file
6. Run `yarn docker:up`.
7. Confirm the directory was mounted  by using `yarn docker:sh` and checking if the mount point contains the content you expected to be mounted.

#### Proposed changelog entry for your changes:

* None needed
